src/core/thread/mle.cpp

```diff
@@ -1724,7 +1726,11 @@ ThreadError Mle::HandleDataResponse(const Message &aMessage, const Ip6::MessageI
     // Active Timestamp
     if (Tlv::GetTlv(aMessage, Tlv::kActiveTimestamp, sizeof(activeTimestamp), activeTimestamp) == kThreadError_None)
     {
+        const MeshCoP::Dataset local = mNetif.GetActiveDataset().GetLocal();
+        const MeshCoP::Timestamp *localTimestamp = local.GetTimestamp();
+
         VerifyOrExit(activeTimestamp.IsValid(), error = kThreadError_Parse);
+        VerifyOrExit(update || (localTimestamp && (activeTimestamp.Compare(*localTimestamp) == 0)), ;);
```

Comment:
I see.  Unfortunately, I think the spec text is a bit confusing in this case.  The intended behavior is that the device updates the Network Data, Active Dataset, and Pending Dataset all together.  The word `local` in this case is not the same as `mLocal` in the code.  The spec's use of `local` just means the device's copy of the respective objects.

So relative to OpenThread, this means that receiving an MLE Data Response is used primarily to update the `mNetwork` versions of those objects and they must all happen together or none at all.  If any of those timestamps are greater than the `mLocal` versions, than the `mLocal` versions are updated as well.

Does this make sense?